### PR TITLE
Forgot page

### PR DESCRIPTION
**Pull request by tenisakb**
- validation
- api communication

### CODE REVIEW

File: src/redux/actions/authentication-actions.ts [hunk 1]
```diff
@@ -50,12 +53,16 @@ const ACTIONS = Object.freeze({
     type: RECOVER_ACCOUNT_HANDLE,
     payload: { recoverWords }
```
Comment (EdmundMai) [inline src/redux/actions/authentication-actions.ts]: replace `recoverWords` with `mnemonic`

File: src/redux/epics/authentication-epic.ts [hunk 2]
```diff
@@ -50,4 +51,41 @@ const loginEpic = (action$, state$, dependencies$) =>
     })
   );
 
-export default combineEpics(loginEpic);
+const recoverAccountHandleEpic = (action$, state$, dependencies$) =>
+  action$.pipe(
+    ofType(authenticationActions.RECOVER_ACCOUNT_HANDLE),
+    switchMap(({ payload }) => {
+      const { recoverWords } = payload;
```
Comment (EdmundMai) [inline src/redux/epics/authentication-epic.ts]: replace `recoverWords` with `mnemonic`

File: src/redux/reducers/authentication-reducer.ts [hunk 3]
```diff
@@ -34,6 +35,19 @@ const authenticationReducer = (state = initState, action) => {
...
+        ...state,
+        recoverHandle: handle
+      };
+
+    case authenticationActions.RECOVER_ACCOUNT_HANDLE_RESET:
+      return {
+        ...state,
+        recoverHandle: null
+      };
+
```
Comment (EdmundMai) [inline src/redux/reducers/authentication-reducer.ts]: add tests

File: src/redux/actions/authentication-actions.ts [hunk 4]
```diff
@@ -50,12 +53,16 @@ const ACTIONS = Object.freeze({
...
+  recoverAccountHandleSuccess: ({ handle }) => ({
+    type: RECOVER_ACCOUNT_HANDLE_SUCCESS,
+    payload: { handle }
   }),
   recoverAccountHandleFailure: ({ error }) => ({
     type: RECOVER_ACCOUNT_HANDLE_FAILURE,
     payload: { error }
+  }),
+  recoverAccountHandleReset: () => ({
+    type: RECOVER_ACCOUNT_HANDLE_RESET
```
Comment (EdmundMai) [inline src/redux/actions/authentication-actions.ts]: tests

File: src/components/login/index.tsx [hunk 5]
```diff
@@ -5,16 +5,17 @@ import authenticationActions from "../../redux/actions/authentication-actions";
 import LoginSlide from "./login-slide";
 
 const mapStateToProps = state => ({
-  status: state.authentication.status
+  status: state.authentication.status,
+  recoverHandle: state.authentication.recoverHandle
```
[the remark this answers is not in the document]
Comment (tenisakb) [reply src/components/login/index.tsx]: fixed

File: src/redux/actions/authentication-actions.ts [hunk 6]
```diff
@@ -22,6 +24,7 @@ const ACTIONS = Object.freeze({
   RECOVER_ACCOUNT_HANDLE,
   RECOVER_ACCOUNT_HANDLE_SUCCESS,
   RECOVER_ACCOUNT_HANDLE_FAILURE,
+  RECOVER_ACCOUNT_HANDLE_RESET,
```
[the remark this answers is not in the document]
Comment (tenisakb) [reply src/redux/actions/authentication-actions.ts]: fixed

File: src/redux/actions/authentication-actions.ts [hunk 7]
```diff
@@ -50,12 +53,16 @@ const ACTIONS = Object.freeze({
...
-    type: RECOVER_ACCOUNT_HANDLE_SUCCESS
+  recoverAccountHandleSuccess: ({ handle }) => ({
+    type: RECOVER_ACCOUNT_HANDLE_SUCCESS,
+    payload: { handle }
   }),
   recoverAccountHandleFailure: ({ error }) => ({
     type: RECOVER_ACCOUNT_HANDLE_FAILURE,
     payload: { error }
+  }),
+  recoverAccountHandleReset: () => ({
```
[the remark this answers is not in the document]
Comment (tenisakb) [reply src/redux/actions/authentication-actions.ts]: fixed

File: src/redux/epics/authentication-epic.ts [hunk 8]
```diff
@@ -50,4 +51,41 @@ const loginEpic = (action$, state$, dependencies$) =>
...
+        }
+      } catch (error) {
+        return of(
+          authenticationActions.recoverAccountHandleFailure({
+            error: new Error("Mnemonic words provided was not valid")
+          })
+        );
+      }
+    })
+  );
```
[the remark this answers is not in the document]
Comment (tenisakb) [reply src/redux/epics/authentication-epic.ts]: done

File: src/redux/reducers/authentication-reducer.ts [hunk 9]
```diff
@@ -34,6 +35,19 @@ const authenticationReducer = (state = initState, action) => {
...
       };
 
+    case authenticationActions.RECOVER_ACCOUNT_HANDLE_SUCCESS:
+      const { handle } = action.payload;
+      return {
+        ...state,
+        recoverHandle: handle
+      };
+
+    case authenticationActions.RECOVER_ACCOUNT_HANDLE_RESET:
```
[the remark this answers is not in the document]
Comment (tenisakb) [reply src/redux/reducers/authentication-reducer.ts]: fixed

File: src/redux/actions/authentication-actions.ts (hunk 1 above)
Comment (tenisakb) [reply src/redux/actions/authentication-actions.ts]: fixed

File: src/redux/epics/authentication-epic.ts (hunk 2 above)
Comment (tenisakb) [reply src/redux/epics/authentication-epic.ts]: fixed

File: src/redux/reducers/authentication-reducer.ts (hunk 3 above)
Comment (tenisakb) [reply src/redux/reducers/authentication-reducer.ts]: done

File: src/redux/actions/authentication-actions.ts (hunk 4 above)
Comment (tenisakb) [reply src/redux/actions/authentication-actions.ts]: done

File: src/redux/epics/authentication-epic.test.js [hunk 10]
```diff
@@ -28,8 +28,33 @@ test("loginEpic", done => {
...
+//     expect(actions).toEqual([
+//       authenticationActions.recoverAccountHandleSuccess({
+//         handle:
+//           "4ce17cf018597e77708e25b03ec176bb59bd01e4cc63380f6c9dffc9b8bac0a0609dd6f101df271def959d1ba9609b0647040713efe5fd5e91042701118ce2f9"
+//       }),
+//       push("/login")
+//     ]);
+//     done();
+//    });
+//});
```
Comment (EdmundMai) [inline src/redux/epics/authentication-epic.test.js]: uncomment this and write a test

**Pull request merged**